frontend/src/lib/components/UnitPicker/UnitPicker.tsx

```diff
@@ -27,125 +23,126 @@ const aggregationDisplayMap = aggregationAxisFormatSelectOptions.reduce((acc, op
                         <>
                             {aggregationAxisFormatSelectOptions.map(({ value, label }, index) => (
                                 <LemonButton
                                     key={index}
-                                    onClick={() => handleChange({ format: value, close: true })}
+                                    onClick={() => handleChange({ format: value })}
                                     status="stealth"
                                     active={value === localAxisFormat}
                                     fullWidth
                                 >
                                     {label}
                                 </LemonButton>
                             ))}
-                            {!!featureFlags[FEATURE_FLAGS.CURRENCY_UNITS] && (
-                                <>
-                                    <LemonDivider />
-                                    <PureField label={'prefix:'}>
-                                        <LemonInput
-                                            value={localAxisPrefix}
-                                            onChange={(prefix) => handleChange({ prefix })}
-                                            onPressEnter={(prefix) => handleChange({ prefix, close: true })}
-                                        />
-                                    </PureField>
-                                    <PureField label={'postfix:'}>
-                                        <LemonInput
-                                            value={localAxisPostfix}
-                                            onChange={(postfix) => handleChange({ postfix })}
-                                            onPressEnter={(postfix) => handleChange({ postfix, close: true })}
-                                        />
-                                    </PureField>
-                                </>
-                            )}
+
+                            <>
+                                <LemonDivider />
+                                <LemonButton
+                                    onClick={() => setCustomUnitModal('prefix')}
+                                    status="stealth"
+                                    active={!!filters.aggregation_axis_prefix}
+                                    fullWidth
+                                >
+                                    Prefix
+                                    {!!filters.aggregation_axis_prefix ? `: ${filters.aggregation_axis_prefix}` : ''}
+                                </LemonButton>
+                                <LemonButton
+                                    onClick={() => setCustomUnitModal('postfix')}
+                                    status="stealth"
+                                    active={!!filters.aggregation_axis_postfix}
+                                    fullWidth
+                                >
+                                    Postfix
```

Comment:
Technically pre-defined units are postfixes too, so this could be "Custom postfix` for clarity.
We should also suffix this option with "…", since it requires further user action (providing the postfix in the modal)